MTA-4662: Update containerless CLI docs (#94)
migtools/mta-documentation: docs/topics/proc_running-the-containerless-mta-cli.adoc

* You have the `JAVA_HOME` environmental variable set.
* You have the `JVM_MAX_MEM` system variable set.
+
[NOTE]
====
If you do not set `JVM_MAX_MEM`, the analysis might hang.
====

> .Procedure for Linux

Author: @ibragins & @istein1 - should we change this header top "Procedure for Linux and macOS?

Also, am I ok to merge this and create a new ticket for the windows procedure?

Reviewer: let's merge as is and then think how to adjust document to fit with Windows in a best possible way @anarnold97

> After unpacking the whole zip file to `~/.kantra`, either add that path to the `$PATH` variable or move the `mta-cli` binary to `/usr/bin`.

Reviewer 2: I think we should only mention moving the binary to the PATH, and remove the part about moving it to `/usr/bin`. That path was only used as an example, and does not hold any significance.

> After unpacking the whole zip file to `~/.kantra`, either add that path to the `$PATH` variable or move the `mta-cli` binary to `/usr/bin`.

Author: ```suggestion
After unpacking the whole zip file to `~/.kantra`, add that path to the `$PATH` variable:
```

[NOTE]
====
Moving the `mta-cli` binary to `/usr/bin` requires root privileges, whereas adding that path to `$PATH` variable does not require root privileges.
====

. To extract the dependency zip file to `~/.kantra`, run:
+
[source,terminal,subs="attributes+"]
----

> unzip $HOME/mta-cli.<os>.<arch>.zip -d $HOME/.kantra

Reviewer 2: I will need to check whether or not this command works on windows.

Author: Hi @eemcmullan - please can i double-check?

Or can you share a set of instructions to work on Windows? And I could add another section? 

@istein & @ibragins - please could you help here and if you have instructions for macOS, that would be great?

thanks

Reviewer: @anarnold97 as Mac is *nix compatible, same command might work on Mac as well. 
Windows is a totally different story and I need to investigate how exactly this could be done there. I suppose in most cases on Windows UI will be available and Windows default file manager supports unzip operations, so this file should be just unpacked in any convenient way, for example using File explorer, WinZip/WinRar or whatever user usually is using for such operations.

Author: @ibragins - that was what I suspected

Also, I know with windows, it will involve adding to the *Path* in *System Variables*, and I do not have a windows machine to try it on.

And do the windows users need to run in PowerShell as admin?

Thanks

----

> . For example, to move the `mta-cli` binary to `/usr/bin` using `sudo`, run:

Author: ```suggestion
. For example, to add the `mta-cli` binary to `/usr/bin` as a system wide binary, using `sudo`, run:
```

+
[source,terminal,subs="attributes+"]
----
sudo mv ~/.kantra/mta-cli /usr/bin/
----

> . For example, to add to the `$PATH` variable, run:

Author: ```suggestion
. For example, to add to the global `$PATH` variable, run:
```

.Procedure

. Move the installed {ProductShortName} CLI `.zip` file to your `$PATH`:

> .. For example, using `sudo`:

Reviewer 3: I think using 'sudo' is redundant since you write it in the command below

Reviewer: so, for **Linux**:

- Unzip dependency zip to home folder / .kantra
`unzip $HOME/mta-cli.<os>.<arch>.zip -d $HOME/.kantra`

- Put product binary to `/usr/bin`
`sudo mv ~/.kantra/mta-cli /usr/bin/`

OR

- Add `~/.kantra` path to $PATH variable
`export PATH=$HOME/.kantra:$PATH`
Advantage of second option is that it doesn't require root permissions

+
[source,terminal,subs="attributes+"]
----
$ sudo mv <mta_cli_zip>/<os>-mta-cli /usr/bin
----
.. For example, to your `$PATH`:
+
[source,terminal,subs="attributes+"]
----
$ mv <mta_cli_zip>/<os>-mta-cli <PATH>
----
. Unzip `mta-cli.<os>` zip to the `.kantra` directory inside your `$HOME` directory. For example:
+
[source,terminal,subs="attributes+"]
----

> $HOME/mta-cli.<os>.<arch>.zip -d $HOME/.kantra

Reviewer: `unzip $HOME/mta-cli.<os>.<arch>.zip -d $HOME/.kantra`

.Procedure

. Move the installed {ProductShortName} CLI `.zip` file to your `$PATH`:
.. For example, using `sudo`:
+
[source,terminal,subs="attributes+"]
----
$ sudo mv <mta_cli_zip>/<os>-mta-cli /usr/bin
----
.. For example, to your `$PATH`:
+
[source,terminal,subs="attributes+"]
----
$ mv <mta_cli_zip>/<os>-mta-cli <PATH>
----

> . Move the requirements to the `.kantra` directory:

Author: ```suggestion
. Move the extracted requirements from the `mta-cli.<os>` zip to the `.kantra` directory inside your `$HOME` directory:
```

> . Move the extracted requirements from the `mta-cli.<os>` zip to the `.kantra` directory inside your `$HOME` directory:

Author: ```suggestion
. Move the extracted requirements from the `mta-cli.<os>` zip to the `.kantra` directory inside your `$HOME` directory. For example:
```

+
[source,terminal,subs="attributes+"]
----

> $ mv <mta_cli_zip>/<os>-mta-cli /usr/bin

Reviewer: this is correct, only concern is that normally user will require to use `sudo` to copy to `/usr/bin`. Alternative is putting mta-cli into `$HOME/bin` and making sure it is a part of `$path` system variable

Reviewer 2: Maybe it can mention moving the binary to your $PATH, and leave out the `/usr/bin` example. Or make it clear that that command is just an example.

----

. Move the requirements to the `.kantra` directory:
+
[source,terminal,subs="attributes+"]
----

> $ mv $HOME/mta-cli.<os>.<arch> $HOME/.kantra

Reviewer: This is a move command, while there is a zip file that should be unzipped into `$HOME/.kanra`. Am I missing something?

Reviewer 2: Might be better to instead say something along the lines of "move the contents of the cli zip into $HOME/.kantra. Like above, can use this command as an example only.

Author: Updated, thanks

Reviewer: as I wrote, if it is zip - you can't MOVE its content. You should run `unzip` command, so this is not a valid example. Regarding the statement before the command - LGTM.
@anarnold97 please update command with something like 
`unzip $HOME/mta-cli.<os>.<arch>.zip -d $HOME/.kantra`

Reviewer 3: LGTM as far as I can tell.
I'll defer to @ibragins for a final review and validate all the commands.

Author: thanks @ilanit

just wondering if we should cover the Windows procedure and macOS procedure

Thanks

+
[source,terminal,subs="attributes+"]
----
$ mv <mta_cli_zip>/<os>-mta-cli /usr/bin
----

> . Move the requirements to the `.kantra` directory:

Reviewer 3: Move the requirement to the `.kantra` **home** directory

> . Move the requirements to the `.mta-cli` directory:

Author: ```suggestion
. Move the requirements to the `.kantra` directory:
```

+
[source,terminal,subs="attributes+"]
----

> $ mv $HOME/mta-cli.<os>.<arch> $HOME/.mta-cli

Reviewer 3: The above text says move to .kantra, however command says move to .mta-cli

> $ mv $HOME/mta-cli.<os>.<arch> $HOME/.mta-cli

Author: ```suggestion
$ mv $HOME/mta-cli.<os>.<arch> $HOME/.kantra
```

----
+

> NOTE: The CLI can run with the requirements moved to the directory from which you run an analysis. During the analysis, the CLI checks for the requirements in this directory first, and if it does not find the requirements, it searches for them in the `$HOME/.mta-cli` directory.

Author: ```suggestion
NOTE: The CLI can run with the requirements moved to the directory from which you run an analysis. During the analysis, the CLI checks for the requirements in this directory first, and if it does not find the requirements, it searches for them in the `$HOME/.kantra` directory.
```

> NOTE: The CLI can run with the requirements moved to the directory from which you run an analysis. During the analysis, the CLI checks for the requirements in this directory first, and if it does not find the requirements, it searches for them in the `$HOME/.mta-cli` directory.

Reviewer 3: I think this should remain `$HOME/.kantra` per Dylan's previous comment.
@dymurray , could you please confirm?

Reviewer 4: I'd also need this info for one of my PRs (https://github.com/migtools/mta-documentation/pull/91/files).
